centraldogma/repository_watcher.py

```diff
@@ -146,9 +148,8 @@ def _watch(self, num_attempts_so_far: int) -> None:
                 self.notify_listeners()
                 if not old_latest:
                     self._initial_value_future.set_result(new_latest)
-
                 # Watch again for the next change.
-                self._schedule_watch(0)
+                return 0
```

Comment:
Should we need to return `0` regardless of `new_latest` existence if `self._do_watch()` is normally returned?
```suggestion
            return 0
```

Reply:
@ikhoon Thank you for your suggestion. That means [return num_attempts_so_far + 1](https://github.com/line/centraldogma-python/pull/36/commits/9eab2a9a2a8ffaf3aced6d82769cda11726cb5da#diff-b7f5988067f01971a4c6781db55b0a902dc650fa7d0f4495739b489b27244c3bR177) is only executed when exception is raised. And, it seems different with @hexoul [opinion](https://github.com/line/centraldogma-python/pull/36#discussion_r939637874). As his suggestion, I had a plan to make an issue to discuss about this after this PR is merged, but maybe we need to discuss now to be on the same page. Please let me know the next action for me.

Reply:
For me, it is not that different. A main point of the comment you referred was about no fallback. It also makes sense for me because normal case does not mean to retry and a backoff is generally used to retry. Even though I didn't find references about the relationship between `304` and a need to retry, I agreed with @ikhoon.

Reply:
304 is considered a successful response so `_DELAY_ON_SUCCESS_MILLIS` is used for the next scheduling. As a similar approach, Central Dogma Java client resets `num_attempts_so_far` to `0` for non-exceptional responses.
https://github.com/line/centraldogma/blob/e93bc5a04eb3d38a0f315d9c5fdb0db3b0af1f52/client/java/src/main/java/com/linecorp/centraldogma/client/AbstractWatcher.java#L256-L257
https://github.com/line/centraldogma/blob/e93bc5a04eb3d38a0f315d9c5fdb0db3b0af1f52/client/java/src/main/java/com/linecorp/centraldogma/client/AbstractWatcher.java#L294

Reply:
@hexoul 
OK. Thanks for your explaination!

@ikhoon 
Your suggestion has been committed. Could you please check again? Thanks in advance. https://github.com/line/centraldogma-python/pull/36/commits/a830dc267fc61a77f3370ac1d4da33271f9cfb40